salt/utils/vmware.py

```diff
@@ -702,33 +734,43 @@ def wait_for_task(task, instance_name, task_type, sleep_seconds=1, log_level='de
         The task to wait for.
 
     instance_name
-        The name of the ESXi host, vCenter Server, or Virtual Machine that the task is being run on.
+        The name of the ESXi host, vCenter Server, or Virtual Machine that
+        the task is being run on.
 
     task_type
         The type of task being performed. Useful information for debugging purposes.
 
     sleep_seconds
-        The number of seconds to wait before querying the task again. Defaults to ``1`` second.
+        The number of seconds to wait before querying the task again.
+        Defaults to ``1`` second.
 
     log_level
-        The level at which to log task information. Default is ``debug``, but ``info`` is also supported.
+        The level at which to log task information. Default is ``debug``,
+        but ``info`` is also supported.
     '''
     time_counter = 0
     start_time = time.time()
     while task.info.state == 'running' or task.info.state == 'queued':
         if time_counter % sleep_seconds == 0:
-            msg = '[ {0} ] Waiting for {1} task to finish [{2} s]'.format(instance_name, task_type, time_counter)
+            msg = '[ {0} ] Waiting for {1} task to finish [{2} s]'.format(
+                instance_name, task_type, time_counter)
             if log_level == 'info':
                 log.info(msg)
             else:
                 log.debug(msg)
         time.sleep(1.0 - ((time.time() - start_time) % 1.0))
         time_counter += 1
+    log.trace('task = {0}, task_type = {1}'.format(task,
+                                                   task.__class__.__name__))
     if task.info.state == 'success':
-        msg = '[ {0} ] Successfully completed {1} task in {2} seconds'.format(instance_name, task_type, time_counter)
+        msg = '[ {0} ] Successfully completed {1} task in {2} seconds'.format(
+            instance_name, task_type, time_counter)
         if log_level == 'info':
             log.info(msg)
         else:
             log.debug(msg)
+        # task is in a successful state
+        return task.info.result
     else:
-        raise Exception(task.info.error)
+        # task is in an error state
+        raise task.info.error
```

Comment:
No, this can return any VMware fault (vim.fault.VimFault and sub classes). Because this method is used internally, by other methods in this module, I thought it better to raise the exception as is and handle it properly in the calling code.